jerry-core/ecma/operations/ecma-arraybuffer-object.c

```diff
@@ -0,0 +1,128 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ecma-alloc.h"
+#include "ecma-builtins.h"
+#include "ecma-exceptions.h"
+#include "ecma-gc.h"
+#include "ecma-globals.h"
+#include "ecma-helpers.h"
+#include "ecma-objects.h"
+#include "ecma-objects-general.h"
+#include "ecma-arraybuffer-object.h"
+#include "jmem-heap.h"
+
+#ifndef CONFIG_DISABLE_ARRAYBUFFER_BUILTIN
+
+/** \addtogroup ecma ECMA
+ * @{
+ *
+ * \addtogroup ecmaarraybufferobject ECMA ArrayBuffer object related routines
+ * @{
+ */
+
+/**
+ * ArrayBuffer object creation operation.
+ *
+ * See also: ES2015 24.1.1.1
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value
+ */
+ecma_value_t
+ecma_op_create_arraybuffer_object (const ecma_value_t *arguments_list_p, /**< list of arguments that
+                                                                         are passed to String constructor */
+                              ecma_length_t arguments_list_len) /**< length of the arguments' list */
+{
+  JERRY_ASSERT (arguments_list_len == 0
+                || arguments_list_p != NULL);
+
+  uint32_t length;
+
+  if (arguments_list_p == 0)
+  {
+    length = 0;
+  }
+  else
+  {
+    ecma_number_t num = ecma_get_number_from_value (ecma_op_to_number (arguments_list_p[0]));
+    length = ecma_number_to_uint32 (num);
+    if (num != ((ecma_number_t) length))
+    {
+      return ecma_raise_range_error (ECMA_ERR_MSG ("Invalid ArrayBuffer length."));
+    }
+  }
+
+  ecma_object_t *object_p = ecma_arraybuffer_new_object (length);
+
+  return ecma_make_object_value (object_p);
+} /* ecma_op_create_arraybuffer_object */
+
+/**
+ * Helper function: create arraybuffer object based on the array length
+ *
+ * The struct of arraybuffer object:
+ *   ecma_object_t
+ *   extend_part
+ *   data buffer
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value
+ */
+ecma_object_t *
+ecma_arraybuffer_new_object (ecma_length_t length)
+{
+  ecma_object_t *prototype_obj_p = ecma_builtin_get (ECMA_BUILTIN_ID_ARRAYBUFFER_PROTOTYPE);
+  ecma_object_t *object_p = ecma_create_object (prototype_obj_p,
+                                                sizeof (ecma_extended_object_t) + length,
+                                                ECMA_OBJECT_TYPE_CLASS);
+  ecma_deref_object (prototype_obj_p);
+  ecma_extended_object_t *ext_object_p = (ecma_extended_object_t *) object_p;
+  ext_object_p->u.class_prop.class_id = LIT_MAGIC_STRING_ARRAY_BUFFER_UL;
+  ext_object_p->u.class_prop.value = ecma_make_uint32_value (length);
```

Comment:
I can accept that, but it slightly damaged the readability of our code. Because according to the defination of ecma_extended_object_t, the value should be  "ecma_value_t".
what do you think?